src/tests/purchase.test.ts

```diff
@@ -0,0 +1,36 @@
+import { describe, expect, test } from "vitest";
+import { configurePurchases } from "./base.purchases_test";
+import { APIGetRequest, type GetRequest } from "./test-responses";
+
+describe("purchase", () => {
+  test("purchase loads branding if not preloaded", async () => {
+    const purchases = configurePurchases();
+    const expectedRequest: GetRequest = {
+      url: "http://localhost:8000/rcbilling/v1/branding",
+    };
+    const offerings = await purchases.getOfferings();
+    const packageToBuy = offerings.current?.availablePackages[0];
+    expect(APIGetRequest).not.toHaveBeenCalledWith(expectedRequest);
+    // Currently we hold on the purchase UI, so we add a timeout to not hold the test forever.
+    // We're just checking that the request happened as expected.
+    await Promise.race([
```

Comment:
I had to do this, since the `purchase` method is stuck waiting for the email to be inputted... let me know if you have other suggestions.

Reply:
I think this is the only way to do it for now.
If we would eventually have another method that uses this decorator then we can change it to that one.

But for now it's fine